src/functional/ffront/past_to_itir.py

```diff
@@ -205,6 +217,39 @@ def _construct_itir_domain_arg(
 
         return itir.FunCall(fun=itir.SymRef(id=domain_builtin), args=domain_args)
 
+    def _construct_itir_initialized_domain_arg(
+        self,
+        dim_i: int,
+        dim: Dimension,
+        node_domain: past.Dict,
+    ) -> tuple[itir.Literal, itir.Literal]:
+        try:
+            if node_domain.keys_[dim_i].type.dim == dim:
+                domain_bounds = []
+                for domain_i, domain_input in enumerate(node_domain.values_[dim_i].elts):
+                    if isinstance(domain_input, past.Name):
+                        domain_bounds.append(itir.SymRef(id=domain_input.id))
+                    elif isinstance(domain_input, past.Constant):
+                        domain_bounds.append(
+                            itir.Literal(value=str(domain_input.value), type="int")
+                        )
+                    else:
+                        raise GTTypeError(
+                            f"Expected {domain_i + 1}th domain parameter to be of type {past.Name} or {past.Constant}"
+                            f"but got {type(node_domain.values_[dim_i].elts[0])} "
+                        )
+            else:
+                raise GTTypeError(
+                    f"Dimensions in out field and field domain are not equivalent"
+                    f"Expected {dim}, but got {node_domain.keys_[dim_i].type.dim} "
+                )
+        except Exception:
+            raise GTTypeError(
+                f"Upper and lower bounds could not be determined for {dim_i + 1}th dimension"
+            )
+
+        return domain_bounds[0], domain_bounds[1]
```

Comment:
implementing my comments above step by step, I would expect this function to become more or less a one-liner (probably with an extra assert)
Something like
```
assert len(node_domain.values_[dim_i].elts) == 2 
return (self.visit(bound, as_ref=True) for bound in node_domain.values_[dim_i].elts)